src/main/java/seedu/todo/controllers/ClearController.java

```diff
@@ -160,131 +193,102 @@ else if (dateOn != null) {
-            return String.format("%s and %s", formatTaskMessage(numTasks), formatEventMessage(numEvents));
-        } else if (numTasks != 0) {
-            return formatTaskMessage(numTasks);
-        } else {
-            return formatEventMessage(numEvents);
-        }
-    }
-    
-    private String formatEventMessage (int numEvents) {
-        return String.format("%d %s", numEvents, StringUtil.pluralizer(numEvents, "event", "events"));
-    }
-    
-    private String formatTaskMessage (int numTasks) {
-        return String.format("%d %s", numTasks, StringUtil.pluralizer(numTasks, "task", "tasks"));
-    }
 
     /**
-     * display error message due to invalid clear command
-     * 
-     * @param input
-     *            based on user input
-     * @param parsedDate            
-     *            the date entered by the user      
-     */
-    private void displayErrorMessage(String input, String[] parsedDate, boolean deleteAll, boolean isTask) {
-        String consoleDisplayMessage = String.format("You have entered : %s.",input);
-        String commandLineMessage = COMMAND_WORD;
-        if (!deleteAll) {
-            if (isTask) {
-                commandLineMessage = String.format("%s %s", commandLineMessage, "task");
-            } else {
-                commandLineMessage = String.format("%s %s", commandLineMessage, "event");
-            }
-        }
-        if (parsedDate != null) {
-            if (parsedDate[0] != null) {
-                commandLineMessage = String.format("%s by <date>", commandLineMessage);
-            } else if (parsedDate[1] != null && parsedDate[2] != null) {
-                commandLineMessage = String.format("%s from <date> to <date>", commandLineMessage);
-            } else if (parsedDate[1] != null) {
-                commandLineMessage = String.format("%s from <date>", commandLineMessage);
-            } else {
-                commandLineMessage = String.format("%s to <date>", commandLineMessage);
-            }
-        }
-        Renderer.renderDisambiguation(commandLineMessage, consoleDisplayMessage);
-    }
-    
-    /**
      * clear all tasks and events of the date that exist in the database.
      * 
      * @param TodoListDB
      * @param givenDate
      *            null if parsing failed or Due date for Task or start date for Event
+     * @param deleteAll
+     *            true if no CalendarItem Type provided, false if "task" or "event" keyword found
+     * @param isTask
+     *            true if "task" keyword found, false if "event" keyword found
      */
     private void destroyBySelectedDate(TodoListDB db, LocalDateTime givenDate, boolean deleteAll, boolean isTask) {
         int numTasks = db.getTaskByDate(givenDate).size();
         int numEvents = db.getEventByDate(givenDate).size();
-        if (numTasks == 0 && numEvents == 0) {
+        
+        // no tasks or events are been found
+        if (numTasks == EMPTY && numEvents == EMPTY) {
```

Comment:
Don't really like this kind of extracting of magic numbers, it makes it harder to understand for me. I don't think 0 is considered a magic number, neither is null?

If you really wanted to, a much better option would be to extract a method which checks against 0, but really that's quite unnecessary,